### PR TITLE
Make sure test runs are deterministic

### PR DESCRIPTION
**Pull request by ausi**
To prevent random test failures like https://github.com/contao/contao/pull/2801/checks?check_run_id=2119404288

### CODE REVIEW

File: core-bundle/tests/Routing/RouteProviderTest.php
Comment (ausi) [inline core-bundle/tests/Routing/RouteProviderTest.php]: @aschempp Is it OK to just count upwards here? I’m not sure how this can or should affect the test results.

File: core-bundle/tests/Routing/RouteProviderTest.php
Comment (m-vo) [inline core-bundle/tests/Routing/RouteProviderTest.php]: Why do we need to seed here?

Comment (ausi) [reply core-bundle/tests/Routing/RouteProviderTest.php]: So that the `mt_rand()` in the next line is deterministic.

Comment (m-vo) [reply core-bundle/tests/Routing/RouteProviderTest.php]: Sorry, missed that line completely. :see_no_evil:

**Review by leofeyer — DISMISSED** [review]
Looks good to me. Waiting for @aschempp's answer before merging this.

File: core-bundle/tests/Routing/RouteProviderTest.php
Comment (aschempp) [inline core-bundle/tests/Routing/RouteProviderTest.php:674]: why do we seed the random number generator here? We seed it again in randomBytes?

File: core-bundle/tests/Routing/RouteProviderTest.php
Comment (aschempp) [inline core-bundle/tests/Routing/RouteProviderTest.php:678]: This should be fine, I guess we just need different page IDs on avery `createPage` call. However, this does not really make the tests deterministic, because the page ID will be different depending on the tests run? Maybe we just need to pass an ID to the method or generate it from the arguments?

Comment (ausi) [reply core-bundle/tests/Routing/RouteProviderTest.php:678]: > However, this does not really make the tests deterministic, because the page ID will be different depending on the tests run?

Why? The `pageModelAutoIncrement` member gets reset for every single test.

File: core-bundle/tests/Routing/RouteProviderTest.php
Comment (ausi) [reply core-bundle/tests/Routing/RouteProviderTest.php:674]: For the line `$page->rootSorting = mt_rand();` further down. I think @leofeyer moved the line up here in [`a54cea5` (#2879)](https://github.com/contao/contao/pull/2879/commits/a54cea5e9396faaf85fa615c7f7dc23bfcae8bab)

File: core-bundle/tests/Security/TwoFactor/AuthenticatorTest.php
Comment (Toflar) [inline core-bundle/tests/Security/TwoFactor/AuthenticatorTest.php]: Can we rename this? I know I'm being picky but it's not "random" anymore now. And it doesn't have to be. Let's just name it `generateSecret()` and add a comment about it having to be deterministic for tests or so. Otherwise someone will read this code and go like "wtf?".

Comment (ausi) [reply core-bundle/tests/Security/TwoFactor/AuthenticatorTest.php]: Done in c10dc43c55f7e49aeaf7a8c13deb10bd5d308482

**Review by aschempp — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @ausi.

**Pull request merged**
